src/Components/App/App.tsx

```diff
@@ -11,42 +11,46 @@ const App: FC = () => {
   const [error, setError] = useState('');
   const [warning, setWarning] = useState('');
   const [currentCity, setCurrentCity] = useState<WeatherLocation | null>(null);
-  
+
   const resetAlerts = () => {
     setError('');
     setWarning('');
   }
 
-  let addCity = async (term: string) => {
+  let addCity = async (cityName: string) => {
     resetAlerts();
-    const city = await fetchCityForecast(term);
+    const fetchedCity = await fetchCityForecast(cityName);
 
-    if (!city) {
-      setError(`No location found called '${term}'`);
-    } else if (cities.find(item => item.id === city.id)) {
-      setWarning(`Location '${term}' is already in the list.`);
+    if (!fetchedCity) {
+      setError(`No location found called '${cityName}'`);
+    } else if (cities.find(city => city.id === fetchedCity.id)) {
+      setWarning(`Location '${cityName}' is already in the list.`);
     } else {
-      setCities([city, ...cities]);
+      setCities([fetchedCity, ...cities]);
     }
   };
+
+  let setCityDetails = (city: WeatherLocation) => {
```

Comment:
I'm wondering if city: WeatherLocation | null would help here? A total shot in the dark though =)